### ISSUE TITLE
/* exported */ not support on function?

### ISSUE THREAD

**Issue by Thaina** [eslint/eslint#6591]
I was create function like this and it has error `no-unused-var`

![image](https://cloud.githubusercontent.com/assets/1042507/16552435/26f47b72-41eb-11e6-95e3-d3dcd23bb849.png)

Which this is fine

![image](https://cloud.githubusercontent.com/assets/1042507/16552451/472b29ea-41eb-11e6-8b9b-ac01962e9121.png)

**Comment by eslintbot** [comment]
Hi @Thaina, thanks for the issue. It looks like there's not enough information for us to know how to help you. If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by Thaina** [comment]
I used esline v2.13.1

What I do is easy. Create a js file in project like this
```js
/* exported AnyFunctionName */
/* exported */
function AnyFunctionName() {
    // DoSomething
}
```

Problem is it cause `no-unused-vars` error. Which is because I use this function on another file and don't use in this file anywhere. And eslint don't think function is var

I expect is using `/* exported AnyFunctionName */` or just `/* exported */` above function would make `no-unused-vars` error get away

**Comment by platinumazure** [comment]
A couple of things worth trying:

1. Why are there two `/* exported */` comments? What happens if you remove one?
1. Does the order matter?

**Comment by Thaina** [comment]
@platinumazure Just try to use both way to see what happen. Use only one of them result the same error anyway

I'm not sure order really matter because if `/* exported */` is not directly above function then will it find the closest function?

Anyway as I said, only use one still not work

**Comment by platinumazure** [comment]
@Thaina Apologies, I asked because I'm on a mobile phone and so can't really run ESLint. Thank you for providing that information.

**Comment by Thaina** [comment]
@platinumazure I understand

Also as I was taking screenshot. use `var someVar = function() { }` is not cause any error, with both `/* exported */` and /* exported someVar */ above it work fine. So the problem is not cause from `exported` marking but quite sure that eslint don't see `function A()` as `var` when checking for `exported`

**Comment by ilyavolodin** [comment]
@Thaina We need more information, like parser options and environments you are running in. `exported` doesn't work in Node or CommonJS environments or with `modules` on, for example.

**Comment by Thaina** [comment]
@ilovejary I already setting by the suggestion of exported tag please believe me and read my word

>Also as I was taking screenshot. use var someVar = function() { } is not cause any error, with both /* exported */ and /* exported someVar */ above it work fine

>as I was taking screenshot. use var someVar = function() { } is not cause any error, it work fine

Only `/* exported */` on `function` that don't work

**Comment by platinumazure** [comment]
@Thaina It's not a question of not believing you-- rather, we need more information so we can figure out the best resolution for your problem. Please provide your config file for us so that we can gather that information.

**Comment by Thaina** [comment]
@platinumazure I know you need info but all your suggestion is wrong guess. I already said again and again that the `/* exported */` over normal `var` work properly. I already take screenshot from the start. But you just overlook it and suggest me the wrong way

If you start suggest about what the rule that could be difference in function and var I would accept that you listen to what I have said. But no, you just guessing about `/* exported */` not work which I already said again and again that IT WORK FINE FOR `var`

I am the one who should say it not a question of more information

You need to understand the problem correctly first

**Comment by platinumazure** [comment]
@Thaina Unfortunately, sometimes things get lost in communication. I apologize for any communication errors on my part.

However, if I may offer a piece of advice: You would do well to take into account the possibility of communication issues when writing bug reports on a project. There may be something you know very well, but that project maintainers are unable to pick up immediately from your post. Also, there may be something in your own project's configuration which could be impacting the issue, which you might not be aware of but which we need to know about in triaging issues. In these cases, it is more productive to just provide the information (even if you want to add a caveat of "I'm pretty sure this is not important"), so that everyone can get on the same page. Calling people out and saying they are not taking your issue seriously or "just guessing" is not productive, will not help us resolve your issue, and is honestly just not very nice. Both you and we will have a better experience if we can continue to assume positive intent from each other. Thanks!

**Comment by Thaina** [comment]
@platinumazure I was understand so that I would try to tell you people each person once about my case **That the `/* exported */` is work fine on `var`**

But then you guy just overlook it and repeatly said that `/* exported */` is not work because any setting that would cause it to don't work on `var`

So I should suspect that you guy just not read. or not believe what I have written

That's why I need to go back and first announce that You just not understand the problem. Which you really is. Your guessing that `/* exported */` don't work is the thing that not productive

It might be relate to my setting, but it not about the setting that you guess out while you overlook my detail about **IT WORK FINE FOR** `var` **BUT NOT FOR** `function`

Am I need to repeat again? **IT WORK FINE FOR** `var` **BUT NOT FOR** `function`

So it must not relate to Node or CommonJS or that I have 2 of `/* exported */` tags or the order

because

# **IT WORK FINE FOR** `var` **BUT NOT FOR** `function`

**Comment by Thaina** [comment]
I already file the report and give you some detail. It may not enough but at least it would be productive if you guy just really read it

**Comment by alberto** [comment]
@Thaina we understand you are saying the problem happens just with exported functions and not exported vars, but we need your config to be able to reproduce it. So, if you want this issue to be solved, please provide your config. Otherwise, we will close it as non-reproducible.

**Comment by Thaina** [comment]
@alberto I can give you my config and I will do so

But can you see what you people try to suggest me? It not relate that you guy want my config. You guy just don't look at my screenshot or detail and just randomly said about my setting have node or commonjs. While I already state it clear that IT WORK FINE FOR var BUT NOT FOR function

<details> 
  <summary>anyway this is my config</summary>
```json
{
    "env": {
        "browser": true,
        "es6": true
    },
    "extends": "eslint:recommended",
    "parserOptions": {
        "sourceType": "module"
    },
    "rules": {
        "accessor-pairs": "error",
        "array-bracket-spacing": [
            "error",
            "never"
        ],
        "array-callback-return": "off",
        "arrow-body-style": "error",
        "arrow-parens": [
            "error",
            "always"
        ],
        "arrow-spacing": [
            "error",
            {
                "after": true,
                "before": true
            }
        ],
        "block-scoped-var": "off",
        "block-spacing": "off",
        "brace-style": "off",
        "callback-return": "error",
        "camelcase": [
            "error",
            {
                "properties": "never"
            }
        ],
        "comma-spacing": "off",
        "comma-style": [
            "error",
            "last"
        ],
        "complexity": "off",
        "computed-property-spacing": [
            "error",
            "never"
        ],
        "consistent-return": "off",
        "consistent-this": "off",
        "curly": "off",
        "default-case": "off",
        "dot-location": "error",
        "dot-notation": [
            "error",
            {
                "allowKeywords": true
            }
        ],
        "eol-last": "off",
        "eqeqeq": "off",
        "func-names": "off",
        "func-style": "off",
        "generator-star-spacing": "error",
        "global-require": "error",
        "guard-for-in": "off",
        "handle-callback-err": "error",
        "id-blacklist": "error",
        "id-length": "off",
        "id-match": "error",
        "indent": "off",
        "init-declarations": "off",
        "jsx-quotes": "error",
        "key-spacing": "off",
        "keyword-spacing": "off",
        "linebreak-style": [
            "error",
            "unix"
        ],
        "lines-around-comment": "off",
        "max-depth": "off",
        "max-len": "off",
        "max-lines": "off",
        "max-nested-callbacks": "error",
        "max-params": "off",
        "max-statements": "off",
        "max-statements-per-line": "off",
        "new-parens": "off",
        "newline-after-var": "off",
        "newline-before-return": "off",
        "newline-per-chained-call": "off",
        "no-alert": "off",
        "no-array-constructor": "error",
        "no-bitwise": "off",
        "no-caller": "error",
        "no-catch-shadow": "error",
        "no-confusing-arrow": "error",
        "no-console": "off",
        "no-continue": "off",
        "no-div-regex": "error",
        "no-duplicate-imports": "error",
        "no-else-return": "error",
        "no-empty-function": "off",
        "no-eq-null": "off",
        "no-eval": "off",
        "no-extend-native": "error",
        "no-extra-bind": "error",
        "no-extra-label": "off",
        "no-extra-parens": "off",
        "no-floating-decimal": "error",
        "no-implicit-coercion": "error",
        "no-implicit-globals": "error",
        "no-implied-eval": "error",
        "no-inline-comments": "error",
        "no-inner-declarations": [
            "error",
            "functions"
        ],
        "no-invalid-this": "off",
        "no-iterator": "error",
        "no-label-var": "off",
        "no-lone-blocks": "error",
        "no-lonely-if": "error",
        "no-loop-func": "error",
        "no-magic-numbers": "off",
        "no-mixed-operators": "off",
        "no-mixed-requires": "error",
        "no-multi-spaces": "error",
        "no-multi-str": "error",
        "no-multiple-empty-lines": "error",
        "no-native-reassign": "off",
        "no-negated-condition": "off",
        "no-nested-ternary": "off",
        "no-new": "error",
        "no-new-func": "error",
        "no-new-object": "error",
        "no-new-require": "error",
        "no-new-wrappers": "error",
        "no-octal-escape": "error",
        "no-param-reassign": "off",
        "no-path-concat": "error",
        "no-plusplus": "off",
        "no-process-env": "error",
        "no-process-exit": "error",
        "no-proto": "error",
        "no-prototype-builtins": "off",
        "no-restricted-globals": "error",
        "no-restricted-imports": "error",
        "no-restricted-modules": "error",
        "no-restricted-syntax": "error",
        "no-return-assign": "off",
        "no-script-url": "error",
        "no-self-compare": "error",
        "no-sequences": "off",
        "no-shadow": "off",
        "no-shadow-restricted-names": "error",
        "no-spaced-func": "off",
        "no-sync": "error",
        "no-ternary": "off",
        "no-throw-literal": "off",
        "no-trailing-spaces": "off",
        "no-undef-init": "error",
        "no-undefined": "off",
        "no-underscore-dangle": "off",
        "no-unmodified-loop-condition": "error",
        "no-unneeded-ternary": [
            "error",
            {
                "defaultAssignment": true
            }
        ],
        "no-unsafe-finally": "error",
        "no-unused-expressions": "off",
        "no-use-before-define": "off",
        "no-useless-call": "error",
        "no-useless-computed-key": "error",
        "no-useless-concat": "error",
        "no-useless-constructor": "error",
        "no-useless-escape": "error",
        "no-useless-rename": "error",
        "no-var": "off",
        "no-void": "off",
        "no-warning-comments": "error",
        "no-whitespace-before-property": "error",
        "no-with": "error",
        "object-curly-newline": "off",
        "object-curly-spacing": "off",
        "object-property-newline": "off",
        "object-shorthand": "off",
        "one-var": "off",
        "one-var-declaration-per-line": "off",
        "operator-assignment": "off",
        "operator-linebreak": "error",
        "padded-blocks": "off",
        "prefer-arrow-callback": "off",
        "prefer-const": "error",
        "prefer-reflect": "off",
        "prefer-rest-params": "off",
        "prefer-spread": "off",
        "prefer-template": "off",
        "quote-props": "off",
        "quotes": "off",
        "radix": [
            "error",
            "as-needed"
        ],
        "require-jsdoc": "off",
        "require-yield": "error",
        "rest-spread-spacing": "error",
        "semi": "off",
        "semi-spacing": [
            "error",
            {
                "after": false,
                "before": false
            }
        ],
        "sort-imports": "error",
        "sort-vars": "off",
        "space-before-blocks": "off",
        "space-before-function-paren": "off",
        "space-in-parens": [
            "error",
            "never"
        ],
        "space-infix-ops": "off",
        "space-unary-ops": "error",
        "spaced-comment": "off",
        "strict": "off",
        "template-curly-spacing": "error",
        "unicode-bom": [
            "error",
            "never"
        ],
        "valid-jsdoc": "off",
        "vars-on-top": "off",
        "wrap-iife": "off",
        "wrap-regex": "off",
        "yield-star-spacing": "error",
        "yoda": "off"
    }
}
```
</details>

**Comment by alberto** [comment]
You are using `"sourceType": "module"``
From the [rule docs](http://eslint.org/docs/rules/no-unused-vars#exported):

> Note that /* exported */ has no effect for any of the following:
when the environment is node or commonjs
when parserOptions.sourceType is module
when ecmaFeatures.globalReturn is true

So this is working as expected.

**Comment by Thaina** [comment]
Then how can it work with `var `

That would be something wrong here
